Source/AwsCommonRuntimeKit/io/StreamCore.swift

```diff
@@ -0,0 +1,95 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import AwsCIo
+import AwsCCommon
+
+/// aws_input_stream has acquire and release functions which manage the lifetime of this object.
+class IStreamCore {
+    var rawValue: aws_input_stream
+    let iStreamable: IStreamable
+    var isEndOfStream: Bool = false
+    private let allocator: Allocator
+    private var vtable = aws_input_stream_vtable(seek: doSeek,
+            read: doRead,
+            get_status: doGetStatus,
+            get_length: doGetLength,
+            acquire: { _ = Unmanaged<IStreamCore>.fromOpaque($0!.pointee.impl).retain() },
+            release: { Unmanaged<IStreamCore>.fromOpaque($0!.pointee.impl).release() })
```

Comment:
Nit: More Swifty format

```suggestion
    private var vtable = aws_input_stream_vtable(
        seek: doSeek,
        read: doRead,
        get_status: doGetStatus,
        get_length: doGetLength,
        acquire: { _ = Unmanaged<IStreamCore>.fromOpaque($0!.pointee.impl).retain() },
        release: { Unmanaged<IStreamCore>.fromOpaque($0!.pointee.impl).release() }
    )
```